### ISSUE TITLE
Bug: key-spacing should not apply when type starts on next line

### ISSUE THREAD

**Issue by coyotte508** [typescript-eslint/typescript-eslint#6409]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.9.3&sourceType=module&code=JYOwLgpgTgZghgYwgAgMoHsC2ECS5rxLIDeAUMsgM5YQBqcANgK4QBc5yAPlWFKAOYduIJpgBG0ANykAvqSA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYkACAXAngA4oDG0AlobgLQrzkB2uA9ANaL7XKECGJjAc3RgA2pETRoAe2iQAugF8QCoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
interface SomeInterface {
  someValue:
    | TypeA
    | TypeB;
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/key-spacing": ["error", {align: "value"}],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

The code should not be formatted, as the type is starting on next line

### Actual Result

The type is formatted to this:

```ts
interface SomeInterface {
  someValue:
                        | TypeA
    | TypeB;
}
```

### Additional Info

See https://github.com/ridedott/eslint-config/pull/1708

**Comment by bradzacher** [comment]
The base rule does not apply to multi-line object properties:
https://eslint.org/play/#eyJ0ZXh0IjoiY29uc3QgeCA9IHtcbiAgcHJvcDpcbiAgICB2YWx1ZSxcbn0iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7ImtleS1zcGFjaW5nIjpbImVycm9yIl19LCJlbnYiOnsiZXM2Ijp0cnVlfX19

So our extension should similarly not apply to multi-line type properties.

**Comment by JoshuaKGoldberg** [comment]
@bradzacher I'm lost, does your comment not indicate this should be closed?

**Comment by Josh-Cena** [comment]
@JoshuaKGoldberg The issue is that the rule currently _is_ applied to multiline constructs.

**Comment by JoshuaKGoldberg** [comment]
Oop thank you!
